t/ht_affiliate.t

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use Test::More;
+use File::Spec;
+
+use RightsGlobals;
+
+use MdpConfig;
+use Auth::Auth;
+use Auth::ACL;
+use Access::Rights;
+use Institutions;
+use CGI;
+use Utils;
+
+#---- MONEKYPATCHES
+no warnings 'redefine';
+local *Access::Rights::get_rights_attribute = sub {
+    my $self = shift;
+    my ($C, $id) = @_;
+
+    return ( split(/\./, $id) )[-2];    
+};
+
+local *Access::Rights::get_source_attribute = sub {
+    my $self = shift;
+    my ($C, $id) = @_;
+
+    return ( split(/\./, $id) )[-1];    
+};
+
+$Access::Rights::current_location = 'US';
+local *Access::Rights::_resolve_access_by_GeoIP = sub {
+    my $C = shift;
+    my $required_location = shift;
+    return ( $required_location eq $Access::Rights::current_location ) ? 'allow' : 'deny';
+};
+
+local *Auth::Auth::affiliation_is_hathitrust = sub {
+    return 1;
+};
+
+local *Auth::Auth::auth_sys_is_SHIBBOLETH = sub {
+    return 1;
+};
+
+local *Auth::Auth::affiliation_has_emergency_access = sub {
+    return 0;
+};
+
+local *Auth::Auth::affiliation_is_umich = sub {
+    return 0;
+};
+#---- MONEKYPATCHES
+
+
+my $C = new Context;
+my $cgi = new CGI;
+$C->set_object('CGI', $cgi);
+my $config = new MdpConfig(File::Spec->catdir($ENV{SDRROOT}, 'mdp-lib/Config/uber.conf'),
+                           File::Spec->catdir($ENV{SDRROOT}, 'slip-lib/Config/common.conf'));
+$C->set_object('MdpConfig', $config);
+
+my $auth = Auth::Auth->new($C);
+$C->set_object('Auth', $auth);
+
+mock_institutions($C);
+mock_acls($C);
+
+local %ENV = %ENV;
+$ENV{HTTP_HOST} = q{babel.hathitrust.org};
+$ENV{SERVER_ADDR} = q{141.213.128.185};
+$ENV{SERVER_PORT} = q{443};
+$ENV{REMOTE_USER} = 'user';
+$ENV{eppn} = q{user@umich.edu};
+$ENV{umichCosignFactor} = q{UMICH.EDU};
+$ENV{Shib_Identity_Provider} = Auth::Auth::get_umich_IdP_entity_id();
+$ENV{AUTH_TYPE} = q{shibboleth};
+$ENV{affiliation} = q{member@umich.edu};
+
+sub test_attr {
+    my ( $attr, $source, $location ) = @_;
+    my $id = "test.$attr.$source";
+    $Access::Rights::current_location = $location || 'US';
+
+    my $ar = Access::Rights->new($C, $id);
+    my $status = $ar->check_final_access_status($C, $id);
+    return $status;
+}
+
+my $num_tests = 0;
+
+# US institution
+is(test_attr(1, 1), 'allow', 'ht_affiliate + attr=1 + source=1'); $num_tests += 1;
```

Comment:
I'm not wild about seeing magic numbers here. The least-intrusive modification might be to define e.g. `$ATTR_PD` in `RightsGlobals.pm`. I doubt the current rights model will stick around as-is long term, so I don't think putting in the effort to dynamically load the id -> name mapping from the `attributes` table would be worthwhile now.

Reply:
I added some reverse maps in `RightsGlobals.pm` and updated the script to use the labels.